rclcpp/include/rclcpp/executor.hpp

```diff
@@ -107,11 +109,13 @@ class Executor
     }
   }
 
-  void spin_node_once(rclcpp::node::Node::SharedPtr & node, bool nonblocking = false)
+  template<typename Rep = int64_t, typename Period = std::milli>
+  void spin_node_once(rclcpp::node::Node::SharedPtr & node, std::chrono::duration<Rep,
+    Period> timeout = std::chrono::duration<Rep, Period>::zero())
```

Comment:
For public record:

I think I like this the best:

0: non-blocking
negative values: block indefinitely

although I think in our conversation Dirk may have said 0 should be block indefinitely and negative should be non-blocking.

also, I believe previously there was nothing that blocked indefinitely, by default anything that blocked would block for 1 second